lib/core/stores/local.ts

```diff
@@ -207,6 +210,228 @@ export class LocalPromiseStore implements IPromiseStore {
+  }
+
+  async get(id: string): Promise<Schedule> {
+    const schedule = await this.storage.rmw<Schedule>(id, (s) => {
+      if (isSchedule(s)) {
+        return s;
+      } else {
+        throw new ResonateError(ErrorCodes.NOT_FOUND, "Not found");
+      }
+    });
+
+    if (schedule) {
+      return schedule;
+    }
+    throw new ResonateError(ErrorCodes.NOT_FOUND, "Not found");
+  }
+
+  async search(
+    id: string,
+    tags?: Record<string, string>,
+    limit?: number,
+    cursor?: string | undefined,
+  ): Promise<{ cursor: string; schedules: Schedule[] }> {
+    try {
+      const result = await this.storage.search(id, undefined, undefined, undefined);
+
+      const scheduleList: Schedule[] = [];
+      for await (const item of result) {
+        for await (const elem of item) {
+          if (isSchedule(elem)) {
+            scheduleList.push(elem);
+          }
+        }
+      }
+
+      // Filter schedules based on the provided parameters
+      let filteredSchedules = scheduleList.filter((schedule) => schedule.id === id);
+
+      if (tags) {
+        filteredSchedules = filteredSchedules.filter((schedule) =>
+          Object.entries(tags).every(([key, value]) => schedule.tags && schedule.tags[key] === value),
+        );
+      }
+
+      if (cursor !== undefined) {
+        const cursorIndex = filteredSchedules.findIndex((schedule) => schedule.id === cursor);
+        if (cursorIndex !== -1) {
+          filteredSchedules = filteredSchedules.slice(cursorIndex + 1);
+        }
+      }
+
+      if (limit !== undefined) {
+        filteredSchedules = filteredSchedules.slice(0, limit);
+      }
+
+      const nextCursor = filteredSchedules.length > 0 ? filteredSchedules[filteredSchedules.length - 1].id : null;
+
+      return {
+        cursor: nextCursor ?? "",
+        schedules: filteredSchedules,
+      };
+    } catch (error) {
+      console.error("Error searching schedules:", error);
+      throw new ResonateError(ErrorCodes.UNKNOWN, "Internal error");
+    }
+  }
+
+  private parseCronExpression(cron: string): Date {
+    try {
+      return cronParser.parseExpression(cron).next().toDate();
+    } catch (error) {
+      throw new Error(`Error parsing cron expression: ${error}`);
+    }
+  }
+
+  private validateNextRunTime(nextRunDate: Date): number {
+    const nextRunTime = nextRunDate.getTime();
+    if (isNaN(nextRunTime)) {
+      throw new Error("Invalid next run time");
+    }
+    return nextRunTime;
+  }
+
+  private calculateNextRunTime(cron: string): number | undefined {
+    try {
+      const nextRunDate = this.parseCronExpression(cron);
+      return this.validateNextRunTime(nextRunDate);
+    } catch (error) {
+      console.error(error);
+      return undefined;
+    }
+  }
+}
+
+export class LocalStore {
+  private localPromiseStore: LocalPromiseStore;
+  private localScheduleStore: LocalScheduleStore;
+  constructor(private memoryStorage: MemoryStorage) {
```

Comment:
```suggestion
  constructor(private promiseStore: IPromiseStore, private scheduleStore: IScheduleStore) {
```

Let's have the local store take a generic `IPromiseStore` and `IScheduleStore`, then let's move all the control loop logic down here instead. That way we will have a nice separation of promises and schedules (the schedule store will no longer need to know about the promise store).

Reply:
makes the code look so neat!